### PR TITLE
Relations of fundamental invariants

### PR DESCRIPTION
**Pull request by joschmitt**
I implemented an algorithm by Kemper and Steel to compute the relations between fundamental invariants when they are the union of primary and irreducible secondary invariants which relies on linear algebra and not Gröbner bases.

### CODE REVIEW

File: src/Deprecations.jl [hunk 1]
```diff
@@ -1,3 +1,4 @@
 # Deprecated in 0.10.*
 
 @deprecate automorphisms(x::Graph) automorphism_group_generators(x)
+@deprecate _containement_helper(R::MPolyRing, N::Int, M::Int, I::MPolyIdeal, W::Vector, ord::Symbol) _containment_helper(R, N, M, I, W, ord)
```
Comment (fingolfin) [inline src/Deprecations.jl]: This is an internal only function, isn't it? So I don't think we need to deprecate it; or do you have reason to believe there's external code using it?

File: src/InvariantTheory/affine_algebra.jl [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+# Optional arguments
+Using the arguments `:king` or `:primary_and_secondary` for `algo_gens` selects
+the algorithm for the computation of the fundamental invariants (see
+[`fundamental_invariants`](@ref) for details).
+The argument `:groebner_basis` or `:linear_algebra` for `algo_rels` controls
+which algorithm for the computation of the relations between the fundamental
+invariants is used. With `:groebner_basis`, the relations are computed via the
+standard computation of a kernel of a morphism between multivariate polynomial
+rings. The option `:linear_algebra` uses an algorithm by Kemper and Steel
+[KS99, Section 17.5.5] to compute the relations without the use of Groebner
```
Comment (fingolfin) [inline src/InvariantTheory/affine_algebra.jl]: Could you perhaps change this to use `@cite` (grep through the `docs` dir to see examples; the bib file already contains KS99)

**Review by fingolfin — APPROVED** [review]
Thank you, looks great!

(Two minor suggestions, not important, feel free to ignore)

**Comment by fingolfin** [comment]
Also, the documentation test fail. You could try `Oscar.build_doc(doctest = :fix)` (must be done with Julia 1.6)

File: src/Deprecations.jl (hunk 1 above)
Comment (joschmitt) [reply src/Deprecations.jl]: I just thought it might be best practice. But probably it is overly careful. I remove the `@deprecate`.

File: src/InvariantTheory/affine_algebra.jl (hunk 2 above)
Comment (joschmitt) [reply src/InvariantTheory/affine_algebra.jl]: Yes, thanks, I totally forgot that.

**Comment by thofma** [comment]
Do you mind adding a test where we can quickly check the result? Like a symmetric group:
```
M1 = matrix(QQ, 4, 4, [0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0])
M2 = matrix(QQ, 4, 4, [0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1])
R = invariant_ring(M1, M2)
A, = affine_algebra(R)
@test is_zero(modulus(A))
```

Is there a specific reason the `affine_algebra` function moved? Now the tests are in the "wrong" place.

**Comment by joschmitt** [comment]
I added more tests (including an example from the paper by Kemper and Steel) and fixed a corner case...

> Is there a specific reason the `affine_algebra` function moved? Now the tests are in the "wrong" place.

I personally prefer several small files to one large file and with the code for the relations it felt justified to open a new file. I don't see which tests are in the wrong place now. `affine_algebra` was not tested before, I think?

**Comment by thofma** [comment]
Sorry, I misread something. Everything is fine with respect to the files. My bad.

**Comment by thofma** [comment]
Thanks @joschmitt!
